src/intervals/interval_operations/boolean.jl

```diff
@@ -4,200 +4,331 @@
+    (inf(x) == typemax(T)) & (sup(x) == typemin(T))
+
+function isempty_interval(x::Interval)
+    isnai(x) && return false
+    return isempty_interval(bareinterval(x))
+end
 
-Checks if `x` is the set consisting of a single exactly
-representable float. Any float which is not exactly representable
-does *not* yield a thin interval. Corresponds to `isSingleton` of
-the standard.
 """
-isthin(x::Interval) = inf(x) == sup(x)
+    isentire_interval(x::BareInterval)
+    isentire_interval(x::Interval)
 
+Test whether `x` is the entire real line.
+
+Implement the `isEntire` function of the IEEE Standard 1788-2015 (Section 10.6.3).
 """
-    iscommon(x)
+isentire_interval(x::BareInterval{T}) where {T<:NumTypes} =
+    (inf(x) == typemin(T)) & (sup(x) == typemax(T))
+
+function isentire_interval(x::Interval)
+    isnai(x) && return false
+    return isentire_interval(bareinterval(x))
+end
 
-Checks if `x` is a **common interval**, i.e. a non-empty,
-bounded, real interval.
 """
-iscommon(x::Interval) = !(isentire_interval(x) || isempty_interval(x) || isunbounded(x))
+    isbounded(x::BareInterval)
+    isbounded(x::Interval)
 
+Test whether `x` is empty or has finite bounds.
 """
+isbounded(x::BareInterval) = (isfinite(inf(x)) & isfinite(sup(x))) | isempty_interval(x)
+
+function isbounded(x::Interval)
+    isnai(x) && return false
+    return isbounded(bareinterval(x))
+end
+
+"""
+    isbounded(x::BareInterval)
+    isbounded(x::Interval)
+
+Test whether `x` is not empty and has infinite bounds.
+"""
+isunbounded(x::BareInterval) = !isbounded(x)
+
+function isunbounded(x::Interval)
+    isnai(x) && return false
+    return isunbounded(bareinterval(x))
+end
+
+"""
+    isnai(x::BareInterval)
+    isnai(x::Interval)
+
+Test whether `x` is an NaI.
+"""
+isnai(::BareInterval) = false
+
+isnai(x::Interval) = decoration(x) == ill
+
+"""
+    iscommon(x::BareInterval)
+    iscommon(x::Interval)
+
+Test whether `x` is non-empty and bounded.
+"""
+iscommon(x::BareInterval) = !(isentire_interval(x) | isempty_interval(x) | isunbounded(x))
+
+function iscommon(x::Interval)
+    isnai(x) && return false
+    return iscommon(bareinterval(x))
+end
+
+"""
+    isatomic(x::BareInterval)
     isatomic(x::Interval)
 
-Check whether an interval `x` is *atomic*, i.e. is unable to be split.
-This occurs when the interval is empty, or when the upper bound equals the lower
-bound or the bounds are consecutive floating point numbers.
+Test whether `x` is unable to be split. This occurs when the interval is empty,
+or when the upper bound equals the lower bound or the bounds are consecutive
+floating-point numbers.
 """
-isatomic(x::Interval) = isempty_interval(x) || (inf(x) == sup(x)) || (sup(x) == nextfloat(inf(x)))
+isatomic(x::BareInterval) = isempty_interval(x) | (inf(x) == sup(x)) | (sup(x) == nextfloat(inf(x)))
+
+function isatomic(x::Interval)
+    isnai(x) && return false
+    return isatomic(bareinterval(x))
+end
```

Comment:
Note the following:

```julia
julia> julia> IntervalArithmetic.atomic(Float64, 0.5)
BareInterval{Float64}(0.49999999999999994, 0.5000000000000001)

julia> isatomic(ans)
false
```

I understand the complications to make `atomic` return thiner intervals in a performant way (for `Float64`s it retuns two-eps wide intervals), and I agree with the result of `isatomic` for the interval used. Yet, it seems awkward that an interval (or bareinterval) created by `atomic` returns `false` by `isatomic`.

There is no need to make anything about this; I simply want to share this.